app/src/main/java/fr/free/nrw/commons/upload/PendingUploadsFragment.kt

```diff
@@ -0,0 +1,299 @@
+                val listOfRemoved: MutableList<Contribution> = mutableListOf()
+                val last = sortedContributionsList.last()
+                for (i in sortedContributionsList.indices) {
+                    val current = sortedContributionsList[i]
+                    if (current.transferred == 0L && (current.dateModifiedInMillis() / 100) > (last.dateModifiedInMillis() / 100)){
+                        listOfRemoved.add(current)
+                    }
+                }
+                newContributionList.removeAll(listOfRemoved)
+                newContributionList.addAll(listOfRemoved)
+                val adapter = PendingUploadsAdapter(newContributionList, this)
+                binding.pendingUploadsRecyclerView.setAdapter(adapter)
+                binding.progressTextView.setText((totalUploads-contributionsSize).toString() + "/" + totalUploads + " uploaded")
+                binding.progressBarPending.progress = totalUploads-contributionsSize
+                if (pausedOrQueuedUploads == contributionsSize) {
+                    uploadProgressActivity.setPausedIcon(true)
+                }else{
+                    uploadProgressActivity.setPausedIcon(false)
+                }
+            }
+        }
+    }
+
+    override fun deleteUpload(contribution: Contribution?) {
+        showAlertDialog(
+            requireActivity(),
+            String.format(
+                Locale.getDefault(),
+                getString(R.string.cancelling_upload)
+            ),
+            String.format(
+                Locale.getDefault(),
+                getString(R.string.cancel_upload_dialog)
+            ),
+            String.format(Locale.getDefault(), getString(R.string.yes)),
+            String.format(Locale.getDefault(), getString(R.string.no)),
+            {
+                ViewUtil.showShortToast(context, R.string.cancelling_upload)
+                pendingUploadsPresenter.deleteUpload(contribution, this.requireContext().applicationContext)
+                CommonsApplication.cancelledUploads.add(contribution!!.pageId)
+                resetProgressBar()
+            },
+            {}
+        )
+    }
+
+    companion object {
+        @JvmStatic
+        fun newInstance(param1: String, param2: String) =
+            PendingUploadsFragment().apply {
+                arguments = Bundle().apply {
+                    putString(ARG_PARAM1, param1)
+                    putString(ARG_PARAM2, param2)
+                }
+            }
+    }
+
+    fun restartUploads() {
+        if (contributionsList != null){
+            pendingUploadsPresenter.restartUploads(contributionsList, 0 , this.requireContext().applicationContext)
+        }
+    }
+
+    fun pauseUploads() {
+        if (contributionsList != null){
+            pendingUploadsPresenter.pauseUploads(contributionsList, 0, this.requireContext().applicationContext)
+        }
+    }
+
+    fun deleteUploads(){
```

Comment:
How about generalising this method as well? This is being used in the failed uploads fragment too.